src/query/functions-v2/tests/it/scalars/testdata/control.txt

```diff
@@ -275,3 +275,78 @@ evaluation (internal):
 +--------------+---------------------------------------------------------------------------+
 
 
+ast            : [1,2,3,4]
+raw expr       : array(1_u8, 2_u8, 3_u8, 4_u8)
+checked expr   : array<T0=UInt8><T0, T0, T0, T0>(1_u8, 2_u8, 3_u8, 4_u8)
+optimized expr : [1, 2, 3, 4]
+output type    : Array(UInt8)
+output domain  : [{1..=4}]
+output         : [1, 2, 3, 4]
+
+
+ast            : false in (false, true)
+raw expr       : or(eq(false, false), eq(false, true))
+checked expr   : or<Boolean, Boolean>(eq<Boolean, Boolean>(false, false), eq<Boolean, Boolean>(false, true))
+optimized expr : true
+output type    : Boolean
+output domain  : Unknown
+output         : true
+
+
+ast            : int8_col not in (1, 2, 3, 4, 5, null)
+raw expr       : not(in(ColumnRef(0)::Int8, array(1_u8, 2_u8, 3_u8, 4_u8, 5_u8)))
+checked expr   : not<Boolean>(in<Int16, Array(Int16)>(CAST(ColumnRef(0) AS Int16), CAST(array<T0=UInt8><T0, T0, T0, T0, T0>(1_u8, 2_u8, 3_u8, 4_u8, 5_u8) AS Array(Int16))))
```

Comment:
why did it  turne into `Int16`? @andylokandy

Reply:
because `i16` is the common super type of `u8` and `i8`